src/pages/Blog.tsx

```diff
@@ -1,3 +1,67 @@
-const Blog = () => <>Blog</>;
+import { Box, Typography } from "@mui/material";
+import { useSearchParams } from "react-router-dom";
+import BlogCard from "../components/BlogCard";
+import Pagination from "../components/Pagination";
+import { BLOG_PAGE_SIZE } from "../constants/constants";
+import { useBlogs } from "../services/queries/useBlogs";
+import { formatDate } from "../utils/formatDate";
+
+const Blog: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const currentPage = searchParams.get("page")
+    ? Number(searchParams.get("page"))
+    : 1;
+
+  const { isLoading, blogs, error, totalPages } = useBlogs(
+    BLOG_PAGE_SIZE,
+    currentPage
+  );
+
+  const handlePageChange = (
```

Comment:
will this function be recreated on every page change? maybe memoization could help avoid this?